sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -317,43 +317,11 @@ private object ShuffleStage {
   private def getMapStats(stage: ShuffleQueryStageExec): MapOutputStatistics = {
     assert(stage.resultOption.isDefined, "ShuffleQueryStageExec should" +
       " already be ready when executing OptimizeSkewedPartitions rule")
-    stage.resultOption.get.asInstanceOf[MapOutputStatistics]
+    stage.mapStats
```

Comment:
You already have assert in `QueryStage.mapStats`. This method can be removed now. Although I'd prefer the other way around: not define `mapStats` in `QueryStage` but instead leave the logic in the callers.

Reply:
`mapStats` is called in more than one place. I think we should avoid duplicating the code.